### PR TITLE
Feature/upgrade guac

### PR DESCRIPTION
**Pull request by Mikkelhost**
Updated the version of guac. still need to push the new mysql image to our registry.

You can now copy paste in to the vm without having to open the clipboard.

- [x] Push mysql image to gitlab registry

### CODE REVIEW

**Comment by Mikkelhost** [comment]
Forgot to push before making new branch, hense why other commits has made their way into this PR

File: internal/agent/agent.go [hunk 1]
```diff
@@ -204,7 +204,9 @@ func New(conf *Config) (*Agent, error) {
 		newLabs:    make(chan pb.Lab, 10),
 		State: &State{
 			ExClient: exClient,
-			EnvPool:  &env.EnvPool{},
+			EnvPool: &env.EnvPool{
+				Envs: map[string]env.Environment{},
```
Comment (Lanestolen) [inline internal/agent/agent.go]:
```suggestion
				Envs: map[string]env.Environment{},
				Envs: make(map[string]env.Environment),
```
I think best practice would be to use make to initialize maps

File: internal/agent/environment.go [hunk 2]
```diff
@@ -98,6 +98,15 @@ func (a *Agent) CreateEnvironment(ctx context.Context, req *proto.CreatEnvReques
 	go env.Start(context.TODO())
 
 	// TODO add env to envpool
+	a.State.EnvPool.Em.Lock()
+	a.State.EnvPool.Envs[env.EnvConfig.Tag] = env
+	a.State.EnvPool.Em.Unlock()
+
+	a.State.EnvPool.Em.RLock()
+	for k, _ := range a.State.EnvPool.Envs {
```
Comment (Lanestolen) [inline internal/agent/environment.go]:
```suggestion
	for k := range a.State.EnvPool.Envs {
```

File: internal/environment/guac.go [hunk 3]
```diff
@@ -102,9 +102,9 @@ func (guac *Guacamole) create(ctx context.Context, eventTag string) error {
 	guacdAlias := uuid.New().String()
 	dbAlias := uuid.New().String()
 	containers["web"] = docker.NewContainer(docker.ContainerConfig{
-		Image: "registry.gitlab.com/haaukins/core-utils/guacamole",
+		Image: "guacamole/guacamole:1.4.0",
 		EnvVars: map[string]string{
-			"MYSQL_DATABASE": "guacamole_db",
+			"MYSQL_DATABASE": "guacamole_db?useSSL=false",
```
Comment (Lanestolen) [inline internal/environment/guac.go:105]: Is this just for testing? I know it is host only but isn't it best practice to always use ssl if possible?

**Review by Lanestolen — COMMENTED** [review]
Looks good :)

File: internal/agent/agent.go (hunk 1 above)
Comment (Mikkelhost) [reply internal/agent/agent.go]: Will take into account!

File: internal/agent/environment.go (hunk 2 above)
Comment (Mikkelhost) [reply internal/agent/environment.go]: Are you sure that this only returns the key?

File: internal/environment/guac.go (hunk 3 above)
Comment (Mikkelhost) [reply internal/environment/guac.go:105]: I found no evidence in our previously created guac containers that we are using SSL. But for some reason the new version of Guac requires us to explicitly set the useSSL parameter in order to disable it...

Comment (Mikkelhost) [reply internal/environment/guac.go:105]: And the container is not even exposed to localhost. There is no real sensitive information in the guac databases anyways. So the risk and impact of a compromise is very very low

Comment (Lanestolen) [reply internal/environment/guac.go:105]: The reason is new Database standards :)

Comment (Lanestolen) [reply internal/environment/guac.go:105]: Fair, It is just nice to have these considerations written in the git history :)

File: internal/agent/environment.go (hunk 2 above)
Comment (Lanestolen) [reply internal/agent/environment.go]: Yes :)

**Review by Lanestolen — APPROVED** [review]

**Pull request merged**